integration_tests/conftest.py

```diff
@@ -74,3 +78,198 @@ def pipeline_columns(sim, populations) -> List[str]:
+                - ix
+                + int(d * rerr == binom.pmf(ix, denominator, null_hypothesis_max))
+            )
+            pval = binom.cdf(numerator, denominator, null_hypothesis_min) + binom.sf(
+                denominator - y, denominator, null_hypothesis_max
+            )
+        else:
+            d = binom.pmf(numerator, denominator, null_hypothesis_max)
+            ix = self._binary_search_integers(
+                lambda x1: binom.pmf(x1, denominator, null_hypothesis_min),
+                d * rerr,
+                0,
+                np.floor(null_hypothesis_min * denominator),
+            )
+            # y is the number of terms between 0 and mode that are <= d*rerr.
+            # we need to add a 1 to account for the 0 index.
+            # For comparing this with old behavior, see
+            # tst_binary_srch_for_binom_tst method in test_morestats.
+            y = ix + 1
+            pval = binom.cdf(y - 1, denominator, null_hypothesis_min) + binom.sf(
+                numerator - 1, denominator, null_hypothesis_max
+            )
+
+        pval = min(1.0, pval)
+
+        return pval
+
+    def _binary_search_integers(self, a, d, lo, hi):
+        # Copied from scipy.stats._binomtest._binary_search_for_binom_tst
+        """
+        Conducts an implicit binary search on a function specified by `a`.
+
+        Meant to be used on the binomial PMF for the case of two-sided tests
+        to obtain the value on the other side of the mode where the tail
+        probability should be computed. The values on either side of
+        the mode are always in order, meaning binary search is applicable.
+
+        Parameters
+        ----------
+        a : callable
+        The function over which to perform binary search. Its values
+        for inputs lo and hi should be in ascending order.
+        d : float
+        The value to search.
+        lo : int
+        The lower end of range to search.
+        hi : int
+        The higher end of the range to search.
+
+        Returns
+        -------
+        int
+        The index, i between lo and hi
+        such that a(i)<=d<a(i+1)
+        """
+        while lo < hi:
+            mid = lo + (hi - lo) // 2
+            midval = a(mid)
+            if midval < d:
+                lo = mid + 1
+            elif midval > d:
+                hi = mid - 1
+            else:
+                return mid
+        if a(lo) <= d:
+            return lo
+        else:
+            return lo - 1
+
+
+@pytest.fixture(scope="session")
+def fuzzy_checker() -> FuzzyChecker:
+    tester = FuzzyChecker(
+        # NOTE: This will need to be updated when any new tests with fuzzy asserts
+        # are added.
+        # Do not update this if it is not the only thing that is failing!
+        # Early exits from failing tests can make the number of asserts actually performed
+        # different than what it would be with passing tests.
+        num_comparisons=601,
```

Comment:
Where is this coming from? just a manual calculation of the number of comparisons that you implemented elsewhere? I think an inline note would be helpful and it would be cool if we could somehow do some degree of automation here so we don't have to do manual math on such a large number...

I'm not sure something like this is even possible and don't know exactly what implementation would look like if it is, but would be cool if we could just do manual math on each instance that fuzzy checker is called (like "check address on move", for instance) and then that could get compiled here.